src/components/TextArea/__test__/TextArea.spec.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { TextArea } from '..';
+
+describe('TextArea component', () => {
+    it('renders correctly with default props', () => {
+        render(<TextArea textSize="md" />);
+        const textareaElement = screen.getByRole('textbox');
+        expect(textareaElement).toBeInTheDocument();
+        expect(document.querySelector(".deriv-textarea")).toBeInTheDocument();
+    });
+
+    it('renders label when provided', () => {
+        render(<TextArea textSize="md" label="Test Label" />);
+        const labelElement = screen.getByText('Test Label');
+        expect(labelElement).toBeInTheDocument();
+    });
+
+    it('renders hint when provided', () => {
+        render(<TextArea textSize="md" hint="Test Hint" />);
+        const hintElement = screen.getByText('Test Hint');
+        expect(hintElement).toBeInTheDocument();
+        expect(hintElement.tagName.toLowerCase()).toBe('p');
+    });
+
+    it('displays error state when isInvalid is true', () => {
+        render(<TextArea textSize="md" isInvalid />);
+        expect(document.querySelector(".deriv-textarea--error")).toBeInTheDocument();
+    });
+
+    it('triggers onChange event handler', () => {
+        const handleChange = jest.fn();
+        render(<TextArea textSize="md" onChange={handleChange} />);
+        const textareaElement = screen.getByRole('textbox');
+        fireEvent.change(textareaElement, { target: { value: 'Test input' } });
+        expect(handleChange).toHaveBeenCalledTimes(1);
+    });
```

Comment:
```suggestion
    it('triggers onChange event handler', async () => {
        const handleChange = jest.fn();
        render(<TextArea textSize="md" onChange={handleChange} />);
        const textareaElement = screen.getByRole('textbox');
        await userEvent.type(textareaElement, 'Test input');
        expect(handleChange).toHaveBeenCalledTimes(1);
    });
```
userEvent simulates real user behaviour